File: bachelor-thesis-odms/MongoDB/JavaScript-Node.jsODMs/mongoose-project/models/tpc_h_e/orders-e-with-lineitems-array-as-tags-indexed.js
import mongoose from "mongoose";
mongoose.pluralize(null)

const OrdersEWithLineitemsArrayAsTagsIndexedSchema = new mongoose.Schema({
    // o_orderkey stored as _id
    _id: Number,
    o_orderdate: Date,
    o_lineitems_tags_indexed: {
        type: [mongoose.Schema.Types.Mixed],
        index: true
    }
});

export default mongoose.model("ordersEWithLineitemsArrayAsTagsIndexed", OrdersEWithLineitemsArrayAsTagsIndexedSchema);

File: evaluation/microservice-mongodb-mongodbentities-csharp/Model/Embedded/OrdersEWithLineitemsArrayAsTagsIndexed.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Entities;

namespace MongoDBEntitiesMicroservice.Model.Embedded 
{
    public class OrdersEWithLineitemsArrayAsTagsIndexed : IEntity
    {
        [BsonId]
        public int o_orderkey { get; set; }
        public Date o_orderdate { get; set; }
        public List<object> o_lineitems_tags_indexed { get; set; }

        public OrdersEWithLineitemsArrayAsTagsIndexed() { }

        public OrdersEWithLineitemsArrayAsTagsIndexed(int o_orderkey, Date o_orderdate, List<object> o_lineitems_tags_indexed)
        {
            this.o_orderkey = o_orderkey;
            this.o_orderdate = o_orderdate;
            this.o_lineitems_tags_indexed = o_lineitems_tags_indexed;
        }

        public object GenerateNewID() => throw new NotImplementedException();
        public bool HasDefaultID() => false;
    }
}
